tutorials/future/torch/mnist_tutorial.py

```diff
@@ -0,0 +1,114 @@
+from absl import app, flags
+from easydict import EasyDict
+import numpy as np
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import torchvision
+
+from cleverhans.future.torch.attacks import fast_gradient_method, projected_gradient_descent
+
+FLAGS = flags.FLAGS
+
+
+class LeNet5(torch.nn.Module):
+  """
+  LeNet-5
+  (N, in_channels, 32, 32) in, (N, 120) out
+  """
+
+  def __init__(self, in_channels, padding):
+    super().__init__()
+    self.conv1 = nn.Conv2d(in_channels, 6, 5, padding=padding)
+    self.conv2 = nn.Conv2d(6, 16, 5)
+    self.fc1 = nn.Linear(16 * 5 * 5, 120)
+    self.fc2 = nn.Linear(120, 84)
+    self.fc3 = nn.Linear(84, 10)
+
+  def forward(self, x):
+    x = F.max_pool2d(F.relu(self.conv1(x)), 2)
+    x = F.max_pool2d(F.relu(self.conv2(x)), 2)
+    x = x.view(x.size(0), -1)
+    x = F.relu(self.fc1(x))
+    x = F.relu(self.fc2(x))
+    x = self.fc3(x)
+    return x
+
+
+def ld_mnist():
+  train_transforms = torchvision.transforms.Compose([torchvision.transforms.ToTensor()])
+  test_transforms = torchvision.transforms.Compose([torchvision.transforms.ToTensor()])
+  train_dataset = torchvision.datasets.MNIST(root='./data', train=True, transform=train_transforms, download=True)
+  test_dataset = torchvision.datasets.MNIST(root='./data', train=False, transform=test_transforms, download=True)
+  train_loader = torch.utils.data.DataLoader(train_dataset, batch_size=128, shuffle=True, num_workers=2)
+  test_loader = torch.utils.data.DataLoader(test_dataset, batch_size=128, shuffle=False, num_workers=2)
+  return EasyDict(train=train_loader, test=test_loader)
+
+
+def main(_):
+  device = 'cuda' if torch.cuda.is_available() else 'cpu'
+
+  # Load training and test data
+  data = ld_mnist()
+
+  # Instantiate model, loss, and optimizer for training
+  net = LeNet5(1, 2)
```

Comment:
I think we lost the code to use the GPU when available.
```
if device == 'cuda': 
  net = net.cuda()
```

Reply:
or just

net.to(device)